src/codegen/codegen_aocl.cc

```diff
@@ -28,6 +28,10 @@ runtime::Module BuildAOCL(Array<LoweredFunc> funcs, std::string target_str,
     code = (*f)(code).operator std::string();
   }
 
+  // We want to call "aoc" compiler only when we're running on a x86_64 host.
+  // On ARM SoC hosts, we must use "aocl.aocx" file which is precompiled on
+  // x86_64 hosts.
+#if defined(__x86_64__) || defined(_M_X64)
```

Comment:
Why do we need to separate code by the preprocessor? I think that this cause confusion for users, because `build` implicitly change the behavior by architecture.